sentry_auth_onelogin/views.py

```
@@ -1,55 +1,100 @@
 
-class SamlConfigure(AuthView, SamlMixin):
-    def __init__(self, config):
-        self.config = config
 
+class SelectIdP(AuthView):
     def handle(self, request, helper):
-        access_token = helper.fetch_state('data')['access_token']
-        org_list = self.client.get_org_list(access_token)
-
-        form = ConfigureForm(org_list, request.POST or None)
-        if form.is_valid():
-            org_id = form.cleaned_data['org']
-            org = [o for o in org_list if org_id == str(o['id'])][0]
-            helper.bind_state('org', org)
-            return helper.next_step()
-
-        return self.respond('sentry_auth_onelogin/configure.html', {
-            'form': form,
-            'org_list': org_list,
+        error_value = error_url = False
+        id_or_metadata_url = url = ''
+        if 'action_save' in request.POST:
+            id_or_metadata_url = request.POST['id_or_metadata_url']
+
+            # Get metadata url if an app_id was provided
+            try:
+                id_or_metadata_url = ONELOGIN_METADATA_URL_PREFIX + id_or_metadata_url
+            except:
+                pass
+
+            validate_url = URLValidator()
+            try:
+                validate_url(id_or_metadata_url)
+                url = id_or_metadata_url
+            except:
+                pass
+
+            if url:
+                error_url = True
+                try:
+                    data = OneLogin_Saml2_IdPMetadataParser.parse_remote(url)
+
+                    if data and 'idp' in data:
+                        idp_data = SAML2Provider.extract_idp_data_from_parsed_data(data)
+                        form2 = SAMLForm(idp_data)
+                        if form2.is_valid():
+                            helper.bind_state('idp', idp_data)
+                            helper.bind_state('contact', request.user.email)
+                            return helper.next_step()
+                except:
```

Same here regarding bare except.